File: PythonSortingAlgorithms/selectionSort.py
def selection_sort(data, ascending=True):
    #length of the data
    length = len(data)
    
    #for loop to iterate through the list
    for i in range(length):
        min_index = i
        
        #find the minimum element in the unsorted part of the list
        for j in range(i + 1, length):
            if ascending:
                #ascending order
                if data[j] < data[min_index]:
                    min_index = j
            else:
                #descending order
                if data[j] > data[min_index]:
                    min_index = j

        #swap the found minimum element with the current element
        data[i], data[min_index] = data[min_index], data[i]

    return data

    
#list of numbers from 1 to 20
data = [x for x in range(20, 0, -1)]
#original list
print("Original List:", data)

#ascending sort
sorted_data_ascending = selection_sort(data, ascending=True)
print("Sorted List Ascending:", sorted_data_ascending)

#descending sort
sorted_data_descending = selection_sort(data, ascending=False)
print("Sorted List Descending:", sorted_data_descending)

File: JavaScriptSortingAlgorithms/selectionSort.js
function selectionSort(data, ascending = true) {
    //the length of the input array 'data'
    const length = data.length;

    //for loop to iterate over the elements of the array
    for (let i = 0; i < length; i++) {
        //initialise a variable minIndex to the current index i
        let minIndex = i;

        //nested loop to find the index of the minimum/maximum element
        for (let j = i + 1; j < length; j++) {
            // Check if sorting is in ascending order
            if (ascending) {
                //data[j] is smaller than data[minIndex], update minIndex to j
                if (data[j] < data[minIndex]) {
                    minIndex = j;
                }
            } else {
                //if sorting is in descending order, update minIndex if data[j] is larger
                if (data[j] > data[minIndex]) {
                    minIndex = j;
                }
            }
        }

        //swap the element at i with the element at minIndex
        [data[i], data[minIndex]] = [data[minIndex], data[i]];
    }

    //return the sorted array
    return data;
}

//numbers from 1 to 20 in descending order 
const data = Array.from({ length: 20 }, (_, i) => 20 - i)
console.log("Original List:", data);

//ascending order 
const sortedDataAscending = selectionSort([...data], true);
console.log("Sorted List Ascending:", sortedDataAscending);

//descending order 
const sortedDataDescending = selectionSort([...data], false);
console.log("Sorted List Descending:", sortedDataDescending);
